src/expr/impl/src/scalar/cast.rs

```diff
@@ -241,6 +241,17 @@ fn struct_cast(input: StructRef<'_>, ctx: &Context) -> Result<StructValue> {
     Ok(StructValue::new(fields))
 }
 
+/// Cast array with `source_elem_type` into array with `target_elem_type` by casting each element.
+#[function("cast(anymap) -> anymap", type_infer = "panic")]
```

Comment:
Can you elaborate what `panic` is for?

Reply:
https://github.com/risingwavelabs/risingwave/pull/17986/files#diff-88ebe9953a3bb26edc36108212527ee38f67d5cec76e936e66d0e7cb97435f9b